integration-tests/test_display_name_option.py

```diff
@@ -0,0 +1,80 @@
+import pytest
+import uuid
+import logging
+from conftest import loop_until
+
+logger = logging.getLogger(__name__)
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_register_with_display_name(insights_client, fetch_from_inventory):
+    """Test insights-client --register --display-name SOME_NEW_HOSTNAME
+
+    The current display name should appear in inventory
+    """
+    unique_hostname = f"test-qa.{uuid.uuid4()}.csi-client-tools.example.com"
+
+    status = insights_client.run("--register", "--display-name", unique_hostname)
+    assert loop_until(lambda: insights_client.is_registered)
+    assert (
+        "Successfully registered host" in status.stdout
+    ), "An application should inform about successfull registration"
+
+    assert (
+        f"as {unique_hostname}" in status.stdout
+    ), "App should inform a system is registered with the given hostname"
+    response = fetch_from_inventory()
+    logger.debug(f"response from console {response}")
+
+    record = response["results"][0]
+    assert "display_name" in record.keys()
+    assert unique_hostname == record["display_name"]
+
+
+def test_register_twice_with_different_display_name(
+    insights_client, test_config, fetch_from_inventory, subtests
+):
+    """Try to register a host but with different display-name than set before
+
+    Set new display_name and try to register twice.
+
+    Registering twice, even with a different display_name set, will do nothing.
+    The `register` method does check if the host changed at all,
+    it only checks the machine_id
+    """
+    insights_id = None
+    unique_hostname = f"test-qa.{uuid.uuid4()}.csi-client-tools.example.com"
+    unique_hostname_02 = f"test-qa.{uuid.uuid4()}.csi-client-tools.example.com"
+
+    with subtests.test(msg="the first registration"):
+        status = insights_client.run("--register", "--display-name", unique_hostname)
+        assert "Successfully registered host" in status.stdout
```

Comment:
ditto

Reply:
this is a case when user register a system. He tries to register the system once more, with different display-name. Even this case may not make sense - but it may happen. So I think it is worth automating it.

Reply:
Lets do this: 

1.  Create a jira to create `fetch_from_inventory`  functionality in client-tools, and block this on that card.
2.  remove it from this PR
3. merge this once thats working.

I think the rest of this is fine as of the latest change.